features/collection/thresholding/impl/src/main/java/org/opennms/netmgt/threshd/AbstractThresholdEvaluatorState.java

```diff
@@ -151,15 +156,30 @@ public String toString() {
 
         this.thresholdingSession = thresholdingSession;
         kvStore = getKvStoreForType(stateType, thresholdingSession.getBlobStore());
-        key = String.format("%d-%s-%s-%s-%s-%s", thresholdingSession.getKey().getNodeId(),
+        key = String.format("%d-%s-%s-%s-%s-%s-%d", thresholdingSession.getKey().getNodeId(),
                 thresholdingSession.getKey().getLocation(), threshold.getDsType(),
-                threshold.getDatasourceExpression(), thresholdingSession.getKey().getResource(), threshold.getType());
+                threshold.getDatasourceExpression(), thresholdingSession.getKey().getResource(), threshold.getType(),
+                generateHashForThresholdState(threshold));
 
-        stateTTL = SystemProperties.getInteger("org.opennms.netmgt.threshd.state_ttl",
-                (int) TimeUnit.SECONDS.convert(24, TimeUnit.HOURS));
         initializeState();
     }
 
+    // Multiple threshold levels for trigger/rearm may end up with the same key.
+    // Generate unique hashcode for threshold values.
+    private int generateHashForThresholdState(BaseThresholdDefConfigWrapper threshold) {
+        Hasher hasher = Hashing.murmur3_128().newHasher();
```

Comment:
What happens if we didn't `put` anything to the hasher? Does it blow up or handle that gracefully?

Reply:
That's good question.  It just returns bytes of 0x00.

Reply:
Cool, as long as it doesn't blow up we should be ok.